src/StreamJsonRpc.Tests/JsonRpcWithFatalExceptionsTests.cs

```diff
@@ -0,0 +1,321 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+using Nerdbank;
+using StreamJsonRpc;
+using Xunit;
+using Xunit.Abstractions;
+
+public class JsonRpcWithFatalExceptionsTests : TestBase
+{
+    internal static Exception FaultException;
+    private readonly Server server;
+    private readonly DisposingDuplexStream serverStream;
+    private readonly DisposingDuplexStream clientStream;
+    private readonly DelimitedMessageHandler messageHandler;
+    private readonly JsonRpc clientRpc;
+
+    public JsonRpcWithFatalExceptionsTests(ITestOutputHelper logger)
+        : base(logger)
+    {
+        this.server = new Server();
+        var streams = DisposingDuplexStream.CreateStreams();
+        this.serverStream = streams.Item1;
+        this.clientStream = streams.Item2;
+
+        this.messageHandler = new DisposingMessageHander(this.clientStream, this.serverStream);
+        this.clientRpc = new JsonRpcWithFatalExceptions(this.messageHandler, this.server);
+        this.clientRpc.StartListening();
+    }
+
+    [Fact]
+    public async Task CanInvokeMethodOnServer()
+    {
+        string testLine = "TestLine1" + new string('a', 1024 * 1024);
+        string result1 = await this.clientRpc.InvokeAsync<string>(nameof(Server.ServerMethod), testLine);
+        Assert.Equal(testLine + "!", result1);
+    }
+
+    [Fact]
+    public async Task StreamsStayOpenIfCancelledTaskReturned()
+    {
+        var ex = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => this.clientRpc.InvokeAsync(nameof(Server.ServerMethodThatReturnsCancelledTask)));
+        Assert.Equal(CancellationToken.None, ex.CancellationToken);
+
+        // Assert MessageHandler and Stream objects are not disposed
+        Assert.False(((DisposingMessageHander)this.clientRpc.MessageHandler).IsDisposed);
+        Assert.False(this.serverStream.IsDisposed);
+        Assert.False(this.clientStream.IsDisposed);
```

Comment:
Verifying both the message handler and the streams seems overkill, IMO. Particularly since you're actually controlling the MessageHandler being used. I would probably just test MessageHandler's disposal. And then confirm that our message handler tests verify that when the MessageHandler is disposed it disposes of the streams.